src/main/python/feature-extractor/server.py

```diff
@@ -20,53 +21,78 @@
 class Service(service_pb2_grpc.FeatureExtractorServicer):
     """Feature Extractor Service"""
 
+    extractors_names = ["identifiers", "literals", "uast2seq", "graphlet"]
+
+    def Extract(self, request, context):
+        """ Extract features using multiple extrators """
+
+        extractors = []
+
+        for name in self.extractors_names:
+            if request.HasField(name):
+                options = getattr(request, name, None)
+                constructor = getattr(self, "_%s_extractor" % name)
+                extractors.append(constructor(options))
```

Comment:
as far as I understand generated grpc code it can't be None if `request.HasField` returns `true`. But I added just check just in case something changes. Thanks!

Reply:
Oh ok, didn't know that! But yup, better be sure by still adding the check 👍